java/core/src/java/org/apache/orc/impl/OrcTail.java

```diff
@@ -70,8 +70,11 @@ public long getFileModificationTime() {
 
   public OrcFile.WriterVersion getWriterVersion() {
     OrcProto.PostScript ps = fileTail.getPostscript();
+    OrcProto.Footer footer = fileTail.getFooter();
+    OrcFile.WriterImplementation writer =
+        OrcFile.WriterImplementation.from(footer.getWriter());
     return (ps.hasWriterVersion()
```

Comment:
That is what I would prefer.  In my mind the file would be corrupt and we should fail.